src/osd/PGBackend.cc

```diff
@@ -617,7 +617,7 @@ int PGBackend::be_scan_list(
   return 0;
 }
 
-bool PGBackend::be_compare_scrub_objects(
+void PGBackend::be_compare_scrub_objects(
```

Comment:
fwiw i think the error return is still nicer for the caller.  otherwise they have to call ss.str() with constructs a std::string and test it for empty.

Reply:
I think the clenaup i'd make is to make the 'error' enum a bool (error = true, error = false), and/or maybe rename 'found_error'.

Reply:
You are right I could use the value only to replace the ss.empty() check.  This does make me realize that when osd_distrust_data_digest is set and we clear the error the console log will still output the mismatch message that came from be_compare_scrub_objects().

Reply:
It probably still makes sense to log a message that the repair happened.  Maybe that just means there will be 2 messages, one mismatch, and then another repair message, in the console log?